sql/sql_show.cc

```diff
@@ -4116,7 +4131,7 @@ ST_FIELD_INFO processlist_fields_info[] = {
     {"HOST", HOST_AND_PORT_LENGTH - 1, MYSQL_TYPE_STRING, 0, 0, "Host", 0},
     {"DB", NAME_CHAR_LEN, MYSQL_TYPE_STRING, 0, 1, "Db", 0},
     {"COMMAND", 16, MYSQL_TYPE_STRING, 0, 0, "Command", 0},
-    {"TIME", 7, MYSQL_TYPE_LONG, 0, 0, "Time", 0},
+    {"TIME", MAX_DOUBLE_STR_LENGTH, MYSQL_TYPE_DOUBLE, 0, 0, "Time", 0},
```

Comment:
Any particular reason for double vs. decimal as per the original diff?

Reply:
I noticed some issues for 5.6:
1. `INFORMATION_SCHEMA.PROCESSLIST` "TIME" column defined as `int(7)`
(i.e. "display width = 7" assumes numbers up to `9999999`) was replaced
with `decimal(9,6)` which can address only numbers from `-999.999999` to
`999.999999`
2. `store_result_field_processlist()` uses the `double` data type in
5.6 to store time
```protocol->store(my_timeval_to_double(&tm_delta), 6, &tmp_str);```
instead of `protocol->store_decimal()` that should be used for `decimal(9,6)`

I fixed these issues in my 8.0 port using:
`'TIME' double NOT NULL DEFAULT '0'` for INFORMATION_SCHEMA.PROCESSLIST